### PR TITLE
Allow cmdmod.exec_code_all to utilize cmdmod.run_all parameters

### LINKED ISSUE #42990: Unable to pass command-line arguments or environment variables to cmdmod.exec_code_all
### Description of Issue/Question
Even though cmdmod.exec_code_all [internaly uses](https://github.com/saltstack/salt/blob/develop/salt/modules/cmdmod.py#L2449) cmdmod.run_all, it's not possible to pass custom cmd arguments, environment variables (or other parameters accepted by run_all method).

I will prepare PR to address this issue.

### Versions Report
2017.7.1

### PR DESCRIPTION
**Pull request by lukasraska**
### What does this PR do?
Allows cmdmod.exec_code_all to use all arguments from cmdmod.run_all (except python_shell)

### What issues does this PR fix or reference?
Fixes #42990 

### Previous Behavior
User was unable to pass custom script cmd arguments, set specific environment variables and others.

### New Behavior
User is able to pass all arguments from cmdmod.run_all

### Tests written?

Yes

### CODE REVIEW

File: salt/modules/cmdmod.py [hunk 1]
```diff
@@ -2446,7 +2452,12 @@ def exec_code_all(lang, code, cwd=None):
     else:
         cmd = [lang, codefile]
 
-    ret = run_all(cmd, cwd=cwd, python_shell=False)
+    if isinstance(args, str):
```
Comment (gtmanfred) [inline salt/modules/cmdmod.py]: this should be six.string_types

**Review by gtmanfred — CHANGES_REQUESTED** [review]
There is just one small change that I have for python3/unicode compatibility.

File: salt/modules/cmdmod.py (hunk 1 above)
Comment (DarkKnightCZ) [inline salt/modules/cmdmod.py]: @gtmanfred Right, missed that. Fixed in another commit

**Review by gtmanfred — APPROVED** [review]

**Comment by cachedout** [comment]
Go Go Jenkins!

**Pull request merged**